Use Case #3 (#1)
ietf-rats/draft-labiod-rats-attester-groups: draft-labiod-rats-attester-groups.md

Approach: We can consider an Attester Group of 1000 aircrafts (same manufacturing brand)

Use Case 2: Automotive domain, a Vehicle with embedded Electronic Control Units (ECUs)

Context: CONNECT EU H2020 project.
The automotive industry is moving to a more hierarchical in-vehicle architecture where ECUs are monitored by Zonal Controllers and these in turn communicate with the Vehicle Computer. This is, for instance, how kinematic data are extracted from the sensors all the way up to the vehicle computer to be encoded into a V2X message. This data need to be associated with Evidence on the integrity of the sensor as a data source and this is where group attestation is an interesting capability. The Attester Group can be formed for hierarchical-based attestation,
like Attester Group of all in-vehicle ECUs or attested group of vehicles within an intersection.

Approach: we can consider an Attester Group of a fleet of 70000 vehicles (same brand). We can also consider an Attester Group of similar ECUs.

Use Case 3: AI computing cluster

> Context:The AI computing cluster is a composite computing environment composed of a group of computing nodes/chips on which a computing task is executed.

Author: ```suggestion
Context: An AI computing cluster is a composite computing environment composed of a group of computing nodes/chips on which one ore more computing tasks are executed.
```

A user or an application/large model provider needs to verify the integrity of the collected measurement/evidence information from the composite computing environment.

> Challenge: The cluster may contain heterogeneous trusted roots, and the composition may be dynamically updated. Repeated attestation is not efficient and can be very expensive.

Author: ```suggestion
Challenge: The cluster may contain heterogeneous trusted roots, and the composition may be dynamically updated. Repeated attestation is not efficient if done without context and can be very expensive.
```

> Approach: We can consider a large group of attesters or a set of group attesters. A group attester corresponding to the computing cluster of the execution of a task may be dynamically created or canceled according to the requirements of the computing task. The remote attestation server/verifier verifies the collected measurement information for the composite computing environment. Remote group attestation should hide the complexity of back-end computing node interaction from customers. Generally, a master node in the group is responsible for communicating with the client, responding to the remote attestation challenge request of the client, collecting measurement information of all the group nodes as a whole, and sending the measurement information to the remote attestation server for appraisal.

Author: ```suggestion
Approach: We can consider a large group of attesters or a set of group attesters. An attester group that maps to the nodes of a cluster that executes a specific task may be dynamically created or dissolved according to the requirements of the computing task. One ore more remote attestation server/verifier appraise collected evidence/measurements for the entire composite computing environment. The intend of remote group attestation is to hide the complexity of that back-end computing node interaction from customers (the Relying Parties), while still being able to assess its trustworthiness. Generally, a master node in the group is responsible for communicating with the Verifiers (or indirectly with the customer if they triggered remote attestation as a Relying Party), responding to the remote attestation challenge request of the client, collecting evidence claims of all group nodes as a whole, and sending the evidence to the verifier for appraisal.
```

> When all computing nodes/chips in a computing attester group are provided by the same vendor or deployed by the same cloud vendor, a unified and centralized dedicated hardware root of trust can be deployed (e.g., hardware security chip, centralized hardware DIE, or BMC) to offload important security functions (secure storage, security monitoring, etc.) to this independent module. The trusted boot and other related measurement information of the group are securely stored on the unified root of trust. During the remote attestation process, the master node of the group collects measurement from this module, and the unified root of trust signs all the measurement. If a single root of trust manages multiple chips, a single point of failure (such as malicious intrusion and system breakdown) of the root of trust affects the security of the entire attester group managed by the root of trust. The unified trusted root should support distributed and pooled design. Multiple roots of trust may work together to enhance overall security and reliability.

Author: ```suggestion
When all computing nodes/chips in a computing attester group are provided by the same vendor or deployed by the same cloud vendor, using a unified and centralized dedicated hardware root of trust can be considered (e.g., hardware security chip, centralized hardware DIE, or BMC) to offload important security functions (secure storage, security monitoring, etc.) to this independent root of trust module. The trusted boot and other related evidence claims of the group are securely stored on that unified root of trust. During the remote attestation procedure, the master node of the group collects claims aggregated and signed already as evidence by the centralized module. If a single root of trust manages multiple chips, a single point of failure (such as malicious intrusion and system breakdown) of the root of trust affects the security of the entire attester group managed by the root of trust. The unified trusted root should support distributed and pooled design. Multiple roots of trust may work together to enhance overall security and reliability.
```

Author: I do not what a pooled design is.

Author: The terms root of trust and module seem to interchangable and that might create confusion if they are perceived as different centralized components.

> In heterogeneous interconnection scenarios where all computing nodes and chips in a computing cluster are provided by different vendors, a unified root of trust may not be deployed. During the group remote attestation process, the master node needs to communicate with each group node to collect its own measurement information. The information is signed by the private key of each node. The master node collects the information, packs the information, and sends it to the remote attestation server for appraisal.

Author: ```suggestion
In heterogeneous interconnection scenarios where all computing nodes and chips in a computing cluster are provided by different vendors, it might not be possible to deploy a unified root of trust. During the group remote attestation procedure, the master node needs to communicate with each group node to collect its individual evidence claims. The node evidence is signed by the private attestation key of each node. The master node collects the information, packs the information, and sends it to the remote verifier for appraisal.
```

The dynamicity of the computing attested group is refected through the following aspects:

> -	the creation and cancellation of a group is dynamically triggered by the life cycle management of computing tasks. The member scale, type, and quantity of measurement information to be collected are dynamically generated and dynamically change. Before performing remote attestation, customers are required to dynamically obtain all related information through the management system interface. Based on this, a template-based remote attestation request message is defined and sent to the master node.

Author: ```suggestion
-	Creation and dissolving of groups is dynamically triggered by the life cycle management of computing tasks the groups execute. The member scale, type, and quantity of evidence claims to be collected are dynamically generated and dynamically change. Before performing remote attestation, customers are required to dynamically obtain all related information through a management system interface. Based on this management information, a template-based remote attestation request message is defined and sent to the master node.
```

> -	According to the dynamic requirements of computing task functions, performance, and to the trustworthy state changes of member nodes, the overall state of the group (including the state change of each existing node, the exit of the existing node from the group, the addition of a new node to the group, the replacement of the existing node by the new node, etc.) is dynamic. These dynamic changes lead to the need for real-time dynamic update of group remote attestation. In this case, incremental update should be supported. That is, updating the measurement value of a single node does not cause the measurement of all or other nodes to be recollected and calculated, resulting in the increased communication and computing load. Especially when the group size is large, these unnecessary load needs to be optimized.

Author: ```suggestion
-	According to the dynamic requirements of computing task functions, performance, and to the trustworthy state changes of member nodes, the overall state of the group (including the state change of each existing node, the exit of the existing node from the group, the addition of a new node to the group, the replacement of the existing node by the new node, etc.) is dynamic. These dynamic changes lead to the need for real-time dynamic update of group remote attestation. Incremental update should also be supported to reduce communication and computing load in large groups.
```

> -	According to the dynamic requirements of computing task functions, performance, and to the trustworthy state changes of member nodes, the overall state of the group (including the state change of each existing node, the exit of the existing node from the group, the addition of a new node to the group, the replacement of the existing node by the new node, etc.) is dynamic. These dynamic changes lead to the need for real-time dynamic update of group remote attestation. In this case, incremental update should be supported. That is, updating the measurement value of a single node does not cause the measurement of all or other nodes to be recollected and calculated, resulting in the increased communication and computing load. Especially when the group size is large, these unnecessary load needs to be optimized.

Author: ```suggestion
-	According to the dynamic requirements of computing task functions, performance, and to the trustworthy state changes of member nodes, the overall state of the group (including the state change of each existing node, the exit of the existing node from the group, the addition of a new node to the group, the replacement of the existing node by the new node, etc.) is dynamic. These dynamic changes lead to the need for real-time dynamic update of group remote attestation. In this case, incremental update should be supported. That is, updating the measurement value of a single node does not cause the measurement of all or other nodes to be recollected and calculated, resulting in the increased communication and computing load. Especially, if the group size is large, these types of redundant evidence claims collections pose unnecessary load that can be avoided.
```

> In the process of group remote attestation, the client can not only integrate the communication key negotiation with the master node, but also support the communication key negotiation with other nodes in the group. Therefore, the key negotiation material is required to be generated by the client and different nodes separately. In addition, the exchange is completed in the request message and result return message of the remote attestation, and the exchange is transmitted to the two ends of each communication session, so that the client and each group node that need to communicate can calculate the session key for mutual communication, so as to implement the subsequent establishment of the security channel.

Author: ```suggestion
In the process of attester group, the client can not only integrate the communication key negotiation with the master node, but also support the communication key negotiation with other nodes in the group. Therefore, the key negotiation material is required to be generated by the client and different nodes separately. In addition, each group node that need to communicate can calculate the session key for mutual communication, so as to implement the subsequent establishment of the security channel.
```

> In the process of group remote attestation, the client can not only integrate the communication key negotiation with the master node, but also support the communication key negotiation with other nodes in the group. Therefore, the key negotiation material is required to be generated by the client and different nodes separately. In addition, the exchange is completed in the request message and result return message of the remote attestation, and the exchange is transmitted to the two ends of each communication session, so that the client and each group node that need to communicate can calculate the session key for mutual communication, so as to implement the subsequent establishment of the security channel.

Author: ```suggestion
In the process of group remote attestation, verifiers can not only conduct key negotiation with the master node, but also support the key negotiation with other nodes inside the group. Therefore, required key negotiation material has to be generated by the verifier and different nodes separately. In addition, the exchange is completed in the request message and result return message of the remote attestation, and the exchange is transmitted to the two ends of each communication session, so that the client and each group node that need to communicate can calculate the session key for mutual communication, so as to implement the subsequent establishment of the security channel.
```

TBH, I cannot follow most of this last paragraph. I am not sure what client means most of the time. It seems to be used interchangeably as verifier, relying party, or customer, which is quite confusing.
